pymongo/change_stream.py

```diff
@@ -101,3 +123,100 @@ def __enter__(self):
+            self._full_pipeline(), self._session, batchSize=self._batch_size,
+            collation=self._collation, maxAwaitTimeMS=self._max_await_time_ms,
+        )
+
+    @property
+    def _database(self):
+        return self._target.database
+
+
+class ChangeStreamDatabase(ChangeStream):
+    """ Class for creating a change stream on all collections in a database.
+    """
+
+    def _run_aggregation_cmd(self, pipeline, session, explicit_session):
+        """Run the full aggregation pipeline for this ChangeStream and return
+        the corresponding CommandCursor.
+        """
+        common.validate_list('pipeline', pipeline)
+        validate_collation_or_none(self._collation)
+        common.validate_non_negative_integer_or_none(
+            "batchSize", self._batch_size
+        )
+
+        cmd = SON([("aggregate", 1),
+                   ("pipeline", pipeline),
+                   ("cursor", {})])
+
+        with self._database.client._socket_for_reads(self._target._read_preference_for(session)) as (sock_info, slave_ok):
+            # Avoid auto-injecting a session: aggregate() passes a session,
+            # aggregate_raw_batches() passes none.
+            result = sock_info.command(
+                self._database.name,
+                cmd,
+                slave_ok,
+                self._target._read_preference_for(session),
+                self._target.codec_options,
+                parse_write_concern_error=True,
+                read_concern=self._target.read_concern,
+                collation=self._collation,
+                session=session,
+                client=self._database.client)
+
+            cursor = result["cursor"]
+            ns = cursor["ns"]
+            _, collname = ns.split(".", 1)
+            aggregation_collection = self._database.get_collection(
+                collname, codec_options=self._target.codec_options,
+                read_preference=self._target._read_preference_for(session),
+                write_concern=self._target.write_concern,
+                read_concern=self._target.read_concern
+            )
+
+            return CommandCursor(
+                aggregation_collection, cursor, sock_info.address,
+                batch_size=self._batch_size or 0,
+                max_await_time_ms=self._max_await_time_ms,
+                session=session, explicit_session=explicit_session
+            )
+
+    def _create_cursor(self):
+        with self._database.client._tmp_session(self._session, close=False) as s:
+            return self._run_aggregation_cmd(
+                self._full_pipeline(),
+                session=s,
+                explicit_session=self._session is not None
+            )
+
+    @property
+    def _database(self):
+        return self._target
+
+
+class ChangeStreamClient(ChangeStreamDatabase):
+    """ Class for creating a change stream on all collections on a cluster. """
+
+    def _full_pipeline(self, options={}):
```

Comment:
We prefer not to use mutable default values for arguments. Consider setting allChangesForCluster another way.